funds.go

```diff
@@ -32,40 +40,127 @@ func deserializeSwapAddressInfo(addressBytes []byte) (*swapAddressInfo, error) {
+	chainPublicKey, chainPrivateKey, err := submarine.GenPublicPrivateKeypair()
+
+	// Send them to the server along with the notification token
+	c := breezservice.NewFundManagerClient(breezClientConnection)
+	ctx, cancel := context.WithTimeout(context.Background(), endpointTimeout*time.Second)
+	defer cancel()
+
+	r, err := c.AddFund(ctx, &breezservice.AddFundRequest{
+		NotificationToken:  notificationToken,
+		PaymentHash:        lightningPaymentHash,
+		ChainPublicKey:     chainPublicKey,
+		LightningNodeId:    lightningPubKey})
 	if err != nil {
+		log.Errorf("Error in AddFund: %v", err)
 		return nil, err
 	}
 
-	invoice, err := lightningClient.AddInvoice(context.Background(), &lnrpc.Invoice{Memo: string(memo), Private: true, Expiry: 60 * 60 * 24 * 30})
+	log.Infof("AddFunds got server address %v", r.Address)
+
+	// Create a script with server's data
+	script, err := submarine.GenSubmarineSwapScript(r.ChainPublicKey, chainPublicKey, lightningPaymentHash, int64(72))
 	if err != nil {
-		log.Criticalf("Failed to call AddInvoice %v", err)
+		log.Errorf("Error in GenSubmarineSwapScript: %v", err)
 		return nil, err
 	}
 
-	c := breezservice.NewFundManagerClient(breezClientConnection)
+	log.Infof("AddFunds local script %v", script)
+
+
+	// Determine which network we are running on
+	var network *chaincfg.Params
+
+	if cfg.Network == "testnet" {
+		network = &chaincfg.TestNet3Params
+	} else if cfg.Network == "simnet" {
+		network = &chaincfg.SimNetParams
+	} else if cfg.Network == "mainnet" {
+		network = &chaincfg.MainNetParams
+	} else {
+		return nil, errors.New("unknown network type " + cfg.Network)
+	}
+
+	log.Infof("AddFunds local network %v", network)
+
+	ourAddress := submarine.GenBase58Address(script, network)
+
+	// Verify we are on the same page
+	if ourAddress != r.Address {
+		return nil, errors.New("base58 address mismatch")
+	}
+
+	// Save everything to DB
+	addressInfo := &swapAddressInfo{
+		Address: r.Address,
+		Paid: false,
+		TransactionConfirmed: false,
+		PaymentHash: lightningPaymentHash,
+		PaymentPreimage: lightningPreimage,
+		OurPubKey: chainPublicKey,
+		OurPrivateKey: chainPrivateKey,
+		PayeePubKey: r.ChainPublicKey,
+		SerializedScript: script,
+	}
+
+	err = saveSwapAddressInfo(addressInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("AddFunds got address %v", r.Address)
+
+	// Create JSON with the script and our private key (in case user wants to do the refund by himself)
+	type ScriptBackup struct {
+		Script string
+		PrivateKey string
+	}
+
+	backup := ScriptBackup{Script: hex.EncodeToString(script), PrivateKey: hex.EncodeToString(chainPrivateKey)}
+	jsonBytes,err := json.Marshal(backup)
+	if err != nil {
+		return nil, err
+	}
+
+	return &data.AddFundReply{Address: r.Address, MaxAllowedDeposit: r.MaxAllowedDeposit, ErrorMessage: r.ErrorMessage, BackupJson: string(jsonBytes[:])}, nil
+}
+
+func SendSwapInvoice(address string, tx string, value int64) error {
```

Comment:
This logic I believe should be triggered from the LND watching the address as soon as a transaction is confirmed regarding this address.
the invoice amount should be taken from the aggregated amount related to this address.
I would suggest to have a simple listener to SubscribeTransaction and whenever a transaction is confirmed re will update the amount on the swapAddressInfo db and mark the address as confirmed.
in the GetPayment just use the regular flow that already pulls the confirmed addresses and request payments from the server.
One thing that needs to be done is to add create payment request at that moment with the amount and add it to the breezservice. GetPaymentRequest as you doing here.